### PR TITLE
feat: Implement Outreach ListMetadata

### PR DESCRIPTION
**Pull request by Jkarage**
Implemented the ListMetdata function for the Outreach connector. Loops through the objects one by one and from the response we generate the metadata field values or errors.
- Include tests for the success and failures.

Running the tests:
Screenshot 2024-05-09 at 17 56 04

### CODE REVIEW

File: outreach/metadata.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		// Constructing the  request url.
+		objURL, err := url.JoinPath(c.BaseURL, obj)
+		if err != nil {
+			return nil, err
+		}
+
+		res, err := c.get(ctx, objURL)
+		if err != nil {
+			objMetadata.Errors[obj] = err
+			continue
```
Comment (eberle1080) [inline outreach/metadata.go:38]:
```suggestion

			continue
```

File: outreach/metadata.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+		}
+
+		// Check nil response body, to avoid panic.
+		if res == nil || res.Body == nil {
+			objMetadata.Errors[obj] = ErrEmptyResponse
+			continue
```
Comment (eberle1080) [inline outreach/metadata.go:44]:
```suggestion

			continue
```

**Review by eberle1080 — APPROVED** [review]

File: outreach/metadata.go (hunk 2 above)
Comment (laurenzlong) [reply outreach/metadata.go:44]: @Jkarage the suggestion wasn't to remove `objMetadata.Errors[obj] = ErrEmptyResponse`, it is to simply add a blank line between it and `continue` can you please add the line back?

Comment (Jkarage) [reply outreach/metadata.go:44]: Noted!!

**Pull request merged**